WcaOnRails/app/views/wfc/panel.html.erb

```diff
@@ -0,0 +1,55 @@
+<% provide(:title, "WFC Panel") %>
+
+<%= render layout: "nav" do %>
+  <h1><%= yield(:title) %></h1>
+
+  <h3>Country bands</h3>
+
+  <%= link_to "Go to country bands", country_bands_path %>
+
+  <h3>Competitions export</h3>
+
+  <%= horizontal_simple_form_for :export, url: "", html: { id: "wfc-export-form" } do |f| %>
+
+    <div class="form-inputs">
+      <%= f.input :from_date, as: :date_picker, input_html: { value: Time.now.strftime("%Y-01-01"), id: "input-from-date" } %>
+      <%= f.input :to_date, as: :date_picker, input_html: { value: Time.now.strftime("%Y-%m-%d"), id: "input-to-date" } %>
+    </div>
+    <div class="col-sm-offset-2 col-sm-10">
+      <%= link_to "Download export", "#", class: "btn btn-primary", id: "download-wfc-export" %>
+    </div>
+    <script>
+      $(function() {
+        $("#download-wfc-export").on("click", function(ev) {
+          ev.preventDefault();
+          const from = $("#input-from-date").val();
+          const to = $("#input-to-date").val();
+          $el = $(this);
+          $el.addClass("disabled");
+          $.ajax({
+            url: '<%= wfc_competitions_export_path %>',
+            method: 'GET',
+            data: {
+              from_date: from,
+              to_date: to,
+            },
+          }).done(function(data) {
+            const fileUrl = URL.createObjectURL(new Blob([data], {type: 'text/csv'}));
+            // Create a hidden link to generate the download
+            const a = document.createElement('a');
+            a.style.display = 'none';
+            a.href = fileUrl;
+            a.download = `wfc-competitions-export-${from}-${to}.csv`;
+            document.body.appendChild(a);
+            a.click();
+            window.URL.revokeObjectURL(fileUrl);
+          }).fail(function(jqXHR, err, message) {
+            alert(message);
+          }).always(function() {
+            $el.removeClass("disabled");
+          });
+        });
+      });
+    </script>
```

Comment:
Is there any reason for using AJAX and dealing with the file manually instead of simply submitting the form and responding with CSV?

Reply:
Yes: loading the file is quite long (because the sql query is expensive for several months); and the issue is that either the button stay enabled during this time (in which case the user could ask twice or more for the same csv), or we disable the button and the user can't download more than one csv.